### DISCUSSION TITLE
Setting up large blocks of data

### DISCUSSION THREAD

**Discussion by GalaxyGateway** [thelsing/knx#208] [Q&A]
Hi, i could use some pointers. I've made a project on esp32 that's interfacing with an alarm panel. Currently i am using esp-knx-ip and do catch all on GA's that i need for:

520 zones states
520 zones alarm states
260 outputs
32 groups
32 groups alarm states
192 rf detectors
64 access zones

These are the max numbers possible, lower numbers depend on the size of the alarm panel used.

The GA simply relates to a zone number in array, i look it up and send it back.

I've read the response in: https://github.com/thelsing/knx/issues/147 where it's not possible to programatically do a catch all because of the nature of having it programmed through ETS. I would like to switch the lib over to this because of the ETS benefit but having to setup 1620 address listeners is proving to be quite a challenge ( in my head ) and don't really see a way how to efficiently do this without running into a bunch of problems for example memory wise.

Would someone have an example or howto handle larger blocks of data like this? Or have a better way to approach this than the BME example?  

All hints are appreciated :-)

**Comment by mumpf** [comment]
Hi,

> but having to setup 1620 address listeners is proving to be quite a challenge

This is funny, I would have more concerns about an ETS Applicaiton with 1620 group objects and all according parameters and how to make it usable for endusers...

But to solve the "catchall" aspect (if I got it correctly): A while ago a did an extension here which is active if you complile with "SMALL_GROUPOBJECTS", which allsows to have 1 callback for any group object change. Your application gets the information, which group object changed, too. Of course your application has to handle this correctly and dispatch to the right handler.

Is this what you are asking for?

Regards,
Waldemar

**Reply by GalaxyGateway** [reply]
Haha well you are right, however sorting the ETS stuff out is a bridge too far for the moment. Still very new to this, scratching the surface of what is presumably much more complex then it appears to be :-) Haven't gotten my head around the XML creation yet ... it's next on the list ...

Thanks for the pointer to the SMALL_GROUPOBJECTS. 

Seems to be a good solution, will take me a couple of days to run some tests with it.

The catch-all refers to a single callback on read command where i sort out what is the GA that's requested to split it, grab the correct array and the data that needs to be in the response. So that the last part of the GA is the key in the array. So it looks like it's exactly what i need!

Thanks!

Tjeerd
